src/mon/MgrMonitor.cc

```diff
@@ -1014,43 +1014,60 @@ bool MgrMonitor::preprocess_command(MonOpRequestRef op)
     f->flush(rdata);
   } else if (prefix == "mgr module ls") {
     if (f) {
+      const auto always_on_modules = map.get_always_on_modules();
+      const auto available_modules = map.available_modules;
+
       f->open_object_section("modules");
-      {
-        f->open_array_section("always_on_modules");
-        for (auto& p : map.get_always_on_modules()) {
-          f->dump_string("module", p);
-        }
-        f->close_section();
+      
+      // always_on_modules
+      f->open_array_section("always_on_modules");
```

Comment:
(not a must for this PR, but if you feel like it:
you can change - in a separate commit - the way sections are opened and closed in the new code to
use the safer RAII wrappers. See for example https://github.com/ceph/ceph/pull/61618 for details).